pkg/clustermesh/namespace/namespace_test.go

```diff
@@ -0,0 +1,308 @@
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			m := &manager{
+				logger: slog.Default(),
+				cfg:    tt.config,
+			}
+			result := m.IsGlobalNamespaceByObject(tt.namespace)
+			assert.Equal(t, tt.expected, result, tt.description)
+		})
+	}
+}
+
+func TestIsGlobalNamespaceByName(t *testing.T) {
+	tests := []struct {
+		name        string
+		config      Config
+		nsName      string
+		namespaces  []*slim_corev1.Namespace
+		expectError bool
+		expected    bool
+		description string
+	}{
+		{
+			name: "namespace exists with annotation true",
+			config: Config{
+				EnableDefaultGlobalNamespace: false,
+			},
+			nsName: "global-ns",
+			namespaces: []*slim_corev1.Namespace{
+				{
+					ObjectMeta: slim_metav1.ObjectMeta{
+						Name: "global-ns",
+						Annotations: map[string]string{
+							annotation.GlobalNamespace: "true",
+						},
+					},
+				},
+			},
+			expectError: false,
+			expected:    true,
+			description: "should return true for annotated global namespace",
+		},
+		{
+			name: "namespace exists with annotation false",
+			config: Config{
+				EnableDefaultGlobalNamespace: true,
+			},
+			nsName: "local-ns",
+			namespaces: []*slim_corev1.Namespace{
+				{
+					ObjectMeta: slim_metav1.ObjectMeta{
+						Name: "local-ns",
+						Annotations: map[string]string{
+							annotation.GlobalNamespace: "false",
+						},
+					},
+				},
+			},
+			expectError: false,
+			expected:    false,
+			description: "should return false for annotated local namespace",
+		},
+		{
+			name: "namespace does not exist",
+			config: Config{
+				EnableDefaultGlobalNamespace: true,
+			},
+			nsName: "non-existent",
+			namespaces: []*slim_corev1.Namespace{
+				{
+					ObjectMeta: slim_metav1.ObjectMeta{
+						Name: "other-ns",
+					},
+				},
+			},
+			expectError: true,
+			expected:    false,
+			description: "should return error when namespace doesn't exist",
+		},
+		{
+			name: "multiple namespaces, find correct one",
+			config: Config{
+				EnableDefaultGlobalNamespace: false,
+			},
+			nsName: "target-ns",
+			namespaces: []*slim_corev1.Namespace{
+				{
+					ObjectMeta: slim_metav1.ObjectMeta{
+						Name: "ns1",
+						Annotations: map[string]string{
+							annotation.GlobalNamespace: "false",
+						},
+					},
+				},
+				{
+					ObjectMeta: slim_metav1.ObjectMeta{
+						Name: "target-ns",
+						Annotations: map[string]string{
+							annotation.GlobalNamespace: "true",
+						},
+					},
+				},
+				{
+					ObjectMeta: slim_metav1.ObjectMeta{
+						Name: "ns3",
+					},
+				},
+			},
+			expectError: false,
+			expected:    true,
+			description: "should find and evaluate correct namespace from multiple",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create mock namespace store
+			store := NewMockNamespaceStore(tt.namespaces...)
+
+			m := &manager{
+				logger: slog.Default(),
+				cfg:    tt.config,
+				store:  store,
+			}
```

Comment:
I'd suggest changing it to something like this to avoid having to mock the entire store API:

```go
var (
	log           = hivetest.Logger(t)
	lc            = hivetest.Lifecycle(t)
	cs, _         = k8sFakeClient.NewFakeClientset(log)
	namespaces, _ = k8s.NamespaceResource(lc, cs, nil)
)

for _, ns := range tt.namespaces {
	_, err := cs.Slim().CoreV1().Namespaces().Create(t.Context(), ns, v1.CreateOptions{})
	require.NoError(t, err)
}

m := newManager(managerParams{
	Logger:     log,
	Lifecycle:  lc,
	Namespaces: namespaces,
	Config:     DefaultConfig,
})
```